getPackages and getModules by Lua api

Edru2 opened the pull request:
#### Brief overview of PR changes/additions
adds functions getPackages() and getModules() in lua api to return a table of installed packages/modules

#### Motivation for adding to Mudlet
fix #4869

#### Other info (issues closed, discussion etc)
When no packages/module is given this just returns nil but I'm not sure if I should change it to return an error message like "no packages/modules installed" and nil or something else (I welcome suggestions)

#### Release post highlight
It is now possible to get a list of installed packages or modules.

Edru2 committed:
getPackages and getModules

Kebap commented:
> When no packages/module is given this just returns nil but I'm not sure if I should change it to return an error message like "no packages/modules installed" and nil or something else (I welcome suggestions)

How about empty table? 

Nil and error I associate with something gone wrong instead.

vadi2 commented:
Let's have a look at how we've done it before + how Lua standard API handles situations like this to resolve

Edru2 committed:
adjust to fit lua api

getAllRoomUserData() and getAreaExits() don't return nil if no data is given + they don't return true if data is given so I adjusted to fit the existing api

Edru2 commented:
The existing api (getAreaExits + getAllRoomUserData) seems to suggest that it should return an empty table. Additionally return true was also not needed.

SlySven commented on src/TLuaInterpreter.cpp:
```diff
@@ -10375,6 +10375,39 @@ int TLuaInterpreter::getModuleSync(lua_State* L)
+        lua_pushnumber(L, counter);
+        lua_pushstring(L, iter.key().toUtf8().constData());
+        lua_settable(L, -3);
+        counter++;
```
This is a teeny bit inefficient. You'd be better to start the value at `0` and then just use the (more efficient)
```cpp
lua_pushnumber(L, ++counter);
```
three lines back, then you do not need this line at all...

SlySven reviewed (commented):
I see you have already (been?) picked-up on the previously suspect `return` values for these two - in these API functions returning an empty table if there is no data is easier for the end-user/scripter to work with... :+1:

Edru2 committed:
suggestion

vadi2 commented:
@Edru2 could you add some examples to it - for example how can one check if a particular package/module is installed?

vadi2 commented:
Thx!
